packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -208,6 +211,7 @@ export function useInnerBlocksProps( props = {}, options = {} ) {
 			'block-editor-block-list__layout',
 			{
 				'has-overlay': hasOverlay,
+				[ layoutClassNames ]: __experimentalLayout,
```

Comment:
It looks like this change might have broken the Columns block layout (the layout classnames don't appear to be output on the wrapper). For that block `__experimentalLayout` doesn't appear to be passed to explicitly down to `useInnerBlocksProps`. I wasn't sure if the current state is that it's required or not for it to be passed down, or if it's just an optional thing that blocks can pass down if they need to be explicit about the layout object being used?

<img width="834" alt="image" src="https://user-images.githubusercontent.com/14988353/197681609-4b7026a6-e2fc-4195-add9-21348ec2c203.png">

Reply:
It's possibly a slightly more rare use case this one: blocks that use layout but don't have any editable controls, so don't expect to pass down a `layout` object, and instead use the default layout object from `block.json`:

https://github.com/WordPress/gutenberg/blob/da9c992b418189c090ec0bd5055960e7f5680f01/packages/block-library/src/columns/block.json#L45-L48

Reply:
Apologies for all the separate comments here! I was curious about how it's hooked together (I couldn't quite remember). This appears to be the line where the fallback to the `block.json` version of the layout object is used:

https://github.com/WordPress/gutenberg/blob/008bb9c94ded2aacb1ea16478f7fb90c7d440215/packages/block-editor/src/hooks/layout.js#L361

Reply:
> It looks like this change might have broken the Columns block layout (the layout classnames don't appear to be output on the wrapper)

Oh, good catch! I knew there was bound to be something wrong 😂 
Yeah it's a good question. I think with Columns it doesn't need layout passing into InnerBlocks because it only uses the default settings for the flex layout type, and doesn't show the controls so isn't user-adjustable.
Hmm I'll have a think about this. It's a bit awful but we need to _not_ add the layout classnames to the inner wrapper for theme back-compat reasons.

Reply:
Just an idea off the top of my head, but would it work to make `__unstableLayoutClassNames` overridable in `useInnerBlocksProps`? E.g. in the hook, use `__unstableLayoutClassNames` if it's passed in within `options`, and fallback to the context? Then in the Group block we could pass an empty string down to `useInnerBlocksProps` for the legacy case... or something like that?

Reply:
> So maybe it should be fixed up in the Columns block?

I'm not sure. It'd be great to _not_ have to pass layout explicitly to innerBlocks in the edit function, and that's something I'd like to look at before we stabilise it.

Reply:
> It'd be great to not have to pass layout explicitly to innerBlocks in the edit function, and that's something I'd like to look at before we stabilise it.

Good thinking 👍

Reply:
OOooh it should work if we just pass an empty string from Group. We shouldn't need any fallback because we just want those classes to _not_ be attached to the inner wrapper.

Reply:
wait, that won't work because innerBlocks is grabbing the classnames directly from context 🤔

Reply:
Yeah I think we'll have to go with something like you're saying @andrewserong! I'm going to make it an unstable prop that can be used to not apply layout classes at all. That way it's more like a skipSerialization type logic, where it won't output anything and the block is responsible for implementing whatever it needs to.

Reply:
That sounds good, and if the block really needs to output its own classnames and pass them down to inner props, it can do that via passing `props.className` anyway 👍